### PR TITLE
Comment section

### PR DESCRIPTION
**Pull request by LucasZibell**
## Summary
The comments are now linked with the backend of the app. It shows the first 4 comments an allows you to create new ones uploading them into the DB

## Trello Card
https://trello.com/c/vFJxjf14/25-comentarios

## ScreenShot
image

### CODE REVIEW

File: React/my-app/src/components/New_Comment/index.js [hunk 1]
```diff
@@ -2,14 +2,20 @@ import React from "react";
 import "./styles.css";
 import { AGREGAR_COMENTARIO, ENVIAR } from "./strings.js";
 import defaultProfile from "../../assets/photos/default-avatar.png";
+import Input from "../Input";
```
Comment (pabloferro) [inline React/my-app/src/components/New_Comment/index.js]: Where are you using this?

File: React/my-app/src/redux/book/actions.js [hunk 2]
```diff
@@ -68,25 +88,52 @@ export const resetGalleryView = () => {
   };
 };
 
-export const getBook = id => {
+export const createNewComment = (bookId, body) => {
   return async dispatch => {
     try {
-      const response = await getBookInfo(id);
+      debugger;
```
Comment (pabloferro) [inline React/my-app/src/redux/book/actions.js]: remove debugger

File: React/my-app/src/redux/book/actions.js [hunk 3]
```diff
@@ -68,25 +88,52 @@ export const resetGalleryView = () => {
...
+      dispatch(uploadingComment(true));
+      const response = await postComment(bookId, body);
       if (responseOK(response)) {
```
Comment (pabloferro) [inline React/my-app/src/redux/book/actions.js]: This is a comment from the previous pr, why do you need to check the response status?
isn't then always an ok response?

File: React/my-app/src/screens/book_detail/index.js [hunk 4]
```diff
@@ -47,6 +52,25 @@ class BookDetailContainer extends Component {
     }
   }
 
+  newComment = event => {
+    event.preventDefault();
+    debugger;
```
Comment (pabloferro) [inline React/my-app/src/screens/book_detail/index.js]: remove debugger

File: React/my-app/src/screens/book_detail/index.js [hunk 5]
```diff
@@ -47,6 +52,25 @@ class BookDetailContainer extends Component {
...
+        createNewComment(bookId, {
+          comment: {
+            book_id: bookId,
+            user_id: window.localStorage.userId,
+            content: commentText
+          }
+        })
+      );
+    }
+    console.log(commentText);
```
Comment (pabloferro) [inline React/my-app/src/screens/book_detail/index.js]: remove log

File: React/my-app/src/screens/book_detail/index.js [hunk 6]
```diff
@@ -47,6 +52,25 @@ class BookDetailContainer extends Component {
...
+    event.preventDefault();
+    debugger;
+    const commentText = event.target.comment.value;
+    const bookId = this.props.match.params.id;
+    if (commentText.length) {
+      this.props.dispatch(
+        createNewComment(bookId, {
+          comment: {
+            book_id: bookId,
+            user_id: window.localStorage.userId,
```
Comment (pabloferro) [inline React/my-app/src/screens/book_detail/index.js:57]: This should be in redux.

File: React/my-app/src/service/books.js [hunk 7]
```diff
@@ -14,3 +14,13 @@ export const getRents = id => {
   api.defaults.headers.common["authorization"] = window.localStorage.token;
   return api.get(`/books/${id}/rents`);
 };
+
+export const getComments = id => {
+  api.defaults.headers.common["authorization"] = window.localStorage.token;
```
Comment (pabloferro) [inline React/my-app/src/service/books.js]: You shouldn't do this in all requests.
Set it once when the user logs in.

**Review by pabloferro — CHANGES_REQUESTED** [review]

File: React/my-app/src/redux/book/actions.js (hunk 3 above)
Comment (LucasZibell) [reply React/my-app/src/redux/book/actions.js]: Nacho told me to evaluate the case in which the server crash, so if it happens the app would be prepare to save the error.

File: React/my-app/src/components/Book/index.js [hunk 8]
```diff
@@ -6,7 +6,7 @@ import { Link } from "react-router-dom";
 const Book = ({ id, imageUrl, title, author }) => (
   <Link to={`/books/${id}`}>
     <div className="book_info">
-      <img className="cover" src={imageUrl || defaultImg} />
+      <img alt={id} className="cover" src={imageUrl || defaultImg} />
```
Comment (pabloferro) [inline React/my-app/src/components/Book/index.js]: Don't use id, alt should represent the image.
Use the book title or an empty string.

File: React/my-app/src/components/NavBar/layout.js [hunk 9]
```diff
@@ -24,13 +24,15 @@ const NavBar = ({
     </div>
     <div className="images_box">
       <img
+        alt=""
```
Comment (pabloferro) [inline React/my-app/src/components/NavBar/layout.js]: alt="notifications"

File: React/my-app/src/components/Suggest/index.js [hunk 10]
```diff
@@ -6,7 +6,12 @@ const Suggest = () => (
   <div className="suggest">
     <div className="suggest_gallery">
       {suggest.map(element => (
-        <img className="tiny_books" src={element.image_url} />
+        <img
+          key={element.id}
+          alt={element.id}
```
Comment (pabloferro) [inline React/my-app/src/components/Suggest/index.js]: don't use id as alt

File: React/my-app/src/config/api.js [hunk 11]
```diff
@@ -5,3 +5,5 @@ export const api = axios.create({
   baseURL: BASE_URL,
   timeout: 15000
 });
+
+api.defaults.headers.common["authorization"] = window.localStorage.token;
```
Comment (pabloferro) [inline React/my-app/src/config/api.js:9]: Add this line too to the login callback

**Review by pabloferro — CHANGES_REQUESTED** [review]

**Review by pabloferro — APPROVED** [review]

**Pull request merged**